### ISSUE TITLE
[Feature Request] Use recency weighting in pretraining

### ISSUE THREAD

**Issue by mbrubeck** [open-spaced-repetition/fsrs4anki#736]
**Which module is related to your feature request?**

Optimizer

**Is your feature request related to a problem? Please describe.**

In my personal Anki collections, initial stability is significantly different for recent cards than for old cards.  However, as far as I can tell, the recency weighting from https://github.com/open-spaced-repetition/fsrs-rs/pull/260 does not apply to the "pretraining" step, so all cards are weighted equally when determining the initial stability params.

I believe that using a higher weight for recent cards in pre-training would improve FSRS accuracy on future cards for me.

**Describe the solution you'd like.**

In the`create_pretrain_data` function in `pre_training.rs`, use `dataset::recency_weighted_fsrs_items` to create weights for the items in the pretraining set. Then, change the `AverageRecall` calculation to compute a weighted average.

**Comment by L-M-Sherlock** [comment]
The initial stability params have been trained further in the training stage with recency weighting. Is it necessary to apply the recency weighting in pretraining stage?

**Comment by Expertium** [comment]
Good question. I think the answer is "test it and see if it improves the results".

**Comment by L-M-Sherlock** [comment]
However, recency weighting is incompatible with the current pretraining method implemented in FSRS-rs...

**Comment by mbrubeck** [comment]
> The initial stability params have been trained further in the training stage with recency weighting. Is it necessary to apply the recency weighting in pretraining stage?

I wasn't aware of that. It does seem like that might be sufficient.

**Comment by L-M-Sherlock** [comment]
I close this issue because I have no idea how to implement it. And the current training stage has tuned the initial stability params with recency weighting.
